internal/services/eventgrid/eventgrid_partner_namespace_resource.go

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package eventgrid
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/eventgrid/2022-06-15/partnernamespaces"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/eventgrid/2022-06-15/partnerregistrations"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.ResourceWithUpdate = EventGridPartnerNamespaceResource{}
+
+type EventGridPartnerNamespaceResource struct{}
+
+type EventGridPartnerNamespaceResourceModel struct {
+	PartnerNamespaceName                string                      `tfschema:"name"`
+	Location                            string                      `tfschema:"location"`
+	ResourceGroup                       string                      `tfschema:"resource_group_name"`
+	InboundIPRules                      []PartnerInboundIpRuleModel `tfschema:"inbound_ip_rule"`
+	LocalAuthEnabled                    bool                        `tfschema:"local_auth_enabled"`
+	PartnerRegistrationFullyQualifiedID string                      `tfschema:"partner_registration_id"`
+	PartnerTopicRoutingMode             string                      `tfschema:"partner_topic_routing_mode"`
+	PublicNetworkAccessEnabled          bool                        `tfschema:"public_network_access_enabled"`
+	Endpoint                            string                      `tfschema:"endpoint"`
+	Tags                                map[string]string           `tfschema:"tags"`
+}
+
+type PartnerInboundIpRuleModel struct {
+	IpMask string `tfschema:"ip_mask"`
+	Action string `tfschema:"action"`
+}
+
+func (EventGridPartnerNamespaceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": &schema.Schema{
```

Comment:
Could you remove `&schema.Schema` from the the map items? This is not required since we've already defined the type in the map's declaration

```suggestion
		"name": {
```